adagrams/game.py

```diff
@@ -1,11 +1,129 @@
+import random
+
+# Function to draw 10 random letters from the letter pool
+
 def draw_letters():
-    pass
+    LETTER_POOL = {
+        'A': 9, 'B': 2, 'C': 2, 'D': 4, 'E': 12, 'F': 2, 'G': 3, 'H': 2, 'I': 9, 'J': 1,
+        'K': 1, 'L': 4, 'M': 2, 'N': 6, 'O': 8, 'P': 2, 'Q': 1, 'R': 6, 'S': 4, 'T': 6,
+        'U': 4, 'V': 2, 'W': 2, 'X': 1, 'Y': 2, 'Z': 1
+    }
+
+ # Create a list of letters with the frequency as specified in the pool
+    letter_list = []
+    for letter, count in LETTER_POOL.items():
+        letter_list.extend([letter] * count)
 
+    # Draw a random sample of 10 letters from the letter list
+    hand = random.sample(letter_list, 10)
+    
+    return hand
+
+# Function to check if the given word can be formed using the available letters
 def uses_available_letters(word, letter_bank):
-    pass
+    word = word.upper()
+        # Create a copy of the letter bank to avoid modifying the original
+
+    letter_bank_copy = letter_bank.copy()
+
+    # Iterate through each letter in the word
+    for letter in word:
+        if letter in letter_bank_copy:
+            letter_bank_copy.remove(letter)
+        else:
+            return False
+
+    # If all letters are in the letter bank copy, the word is valid
+    return True
 
 def score_word(word):
-    pass
+    word = word.upper()
+    letter_scores = {
```

Comment:
Similar to the letter pool above, this dictionary can be declared outside of the function.